File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.js
/**
 * @param {string} colors
 * @param {number[]} neededTime
 * @return {number}
 */
var minCost = function(colors, neededTime) {
  let [ans, maxCost] = [0, neededTime[0]];
  
  for (let i = 0; i < colors.length; i++) {
    if (colors[i] === colors[i - 1]) {
      ans += Math.min(maxCost, neededTime[i]);
      maxCost = Math.max(maxCost, neededTime[i]);
    }
    else maxCost = neededTime[i];
  }
  
  return ans;
};

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        ans, max_cost = 0, neededTime[0]
        for i in range(1, len(colors)):
            if colors[i - 1] == colors[i]:
                ans += min(max_cost, neededTime[i])
                max_cost = max(max_cost, neededTime[i])
            else:
                max_cost = neededTime[i]
        return ans